docs/standard/threading/overview-of-synchronization-primitives.md

```diff
@@ -1,174 +1,167 @@
-  
- [Back to top](#top)  
-  
-<a name="interlocked_operations"></a>   
-## Interlocked Operations  
- Interlocked operations are simple atomic operations performed on a memory location by static methods of the <xref:System.Threading.Interlocked> class. Those atomic operations include addition, increment and decrement, exchange, conditional exchange depending on a comparison, and read operations for 64-bit values on 32-bit platforms.  
-  
+# Overview of synchronization primitives
+
+.NET provides a range of types that you can use to synchronize access to a shared resource or coordinate thread interaction.
+
+## WaitHandle class and lightweight synchronization types
+
+Multiple .NET synchronization primitives derive from the <xref:System.Threading.WaitHandle?displayProperty=nameWithType> class, which encapsulates a native operating system synchronization handle. Those classes include:
+
+- <xref:System.Threading.Mutex?displayProperty=nameWithType>
+- <xref:System.Threading.Semaphore?displayProperty=nameWithType>
+- <xref:System.Threading.EventWaitHandle?displayProperty=nameWithType>
+- <xref:System.Threading.AutoResetEvent?displayProperty=nameWithType>
+- <xref:System.Threading.ManualResetEvent?displayProperty=nameWithType>
+
+As <xref:System.Threading.WaitHandle> derives from <xref:System.MarshalByRefObject?displayProperty=nameWithType>, those types can be used to synchronize the activities of threads across application domain boundaries. Moreover, <xref:System.Threading.Mutex>, <xref:System.Threading.Semaphore>, and <xref:System.Threading.EventWaitHandle> can represent named system synchronization handles, which are visible throughout the operating system and can be used for the inter-process synchronization.
+
+For more information, see the <xref:System.Threading.WaitHandle> API reference.
+
+Lightweight synchronization types don't rely on underlying operating system handles and typically provide better performance. However, they cannot be used for the inter-process synchronization. Use those types for thread synchronization within one application.
+
+Some of those types are alternatives to the types derived from <xref:System.Threading.WaitHandle>. For example, <xref:System.Threading.SemaphoreSlim> is a lightweight alternative to <xref:System.Threading.Semaphore>.
+
+## Synchronization of access to a shared resource
+
+When you synchronize access to a shared resource, you might need to implement one of the two following scenarios:
+
+- At maximum only one thread can access the resource at any given time moment. The access to the resource is mutually exclusive.
+- The number of threads that access the resource at the same time is limited.
```

Comment:
As mentioned in a semaphore note, this is not necessarilly the most common use case.

I'd remove this whole section. It's of limited value and it's somewhat difficult to include all scenarios (e.g. you omitted reader/writer access).